packages/backend/src/models/UserWarehouseCredentials/UserWarehouseCredentialsModel.ts

```diff
@@ -87,6 +88,31 @@ export class UserWarehouseCredentialsModel {
         return this.convertToUserWarehouseCredentials(result);
     }
 
+    async findForProject(
```

Comment:
In the next PR, we will have a table that saves the preferred credentials for a project. So it will check that table first and then fallback to this logic.